### PR TITLE
Checkout: Fix disappearing Privacy Protection issue

### PR DESCRIPTION
**Pull request by gziolo**
From @peterbutler:
> It seems that sometimes a user adds a domain renewal to their cart, then the privacy disappears from it, and the user renews. As a result, the user doesnt pay for privacy

This fix allows multiple products to be added to the cart at once. This should fix also **issue with missing redemption item**.

### Testing
1. Go to Purchases (http://calypso.localhost:3000/purchases).
2. Select expiring domain with privacy protection enabled.
3. Click Renew button, you should see domain and privacy protection in the cart.

### CODE REVIEW

File: client/lib/cart/store/index.js [hunk 1]
```diff
@@ -98,16 +99,8 @@ function update( changeFunction ) {
 	cartAnalytics.recordEvents( previousCart, nextCart );
 }
 
-CartStore.dispatchToken = Dispatcher.register( function( payload ) {
-	var action = payload.action,
-		cartItem;
-
-	if ( action.cartItem ) {
-		cartItem = cartValues.fillInSingleCartItemAttributes(
```
Comment (gziolo) [inline client/lib/cart/store/index.js]: `fillInSingleCartItemAttributes` is invoked in `update` method through `fillInAllCartItemAttributes` method :)
There is no need to call it twice.

**Comment by gziolo** [comment]
This is ready for testing. Pinging @bikedorkjon, @drewblaisdell, @peterbutler and @fabianapsimoes.

**Comment by fabianapsimoes** [comment]
Tested this with the server-side test provided by @peterbutler (10d25-pb / p4jECJ-oX-p2), and it seems to solve the issue :+1: 

Thanks for working on it!

File: client/lib/cart/store/index.js [hunk 2]
```diff
@@ -118,8 +111,8 @@ CartStore.dispatchToken = Dispatcher.register( function( payload ) {
 			update( cartItems.removePrivacyFromAllDomains( CartStore.get() ) );
 			break;
 
-		case UpgradesActionTypes.CART_ITEM_ADD:
-			update( cartItems.add( cartItem ) );
+		case UpgradesActionTypes.CART_ITEMS_ADD:
+			update( flow( ...action.cartItems.map( cartItem => cartItems.add( cartItem ) ) ) );
```
Comment (janm6k) [inline client/lib/cart/store/index.js]: My head hurts from this line :)

But once I understood that `cartItems.add` actually returns function, not cart, than it started to make sense.

**Comment by janm6k** [comment]
Code looks good!

**Comment by scruffian** [comment]
:+1:

File: client/lib/cart/store/index.js (hunk 2 above)
Comment (gziolo) [inline client/lib/cart/store/index.js]: Mine, too :)

I had to use some bits of functional programming combined with ES6 to get this combined into one server update.

If you have any idea how to refactor it, let me know :)

Comment (stephanethomas) [inline client/lib/cart/store/index.js]: You're not the only one :).

Comment (gziolo) [inline client/lib/cart/store/index.js]: It would be good idea to have common pattern for function name that returns another function

**Comment by bikedorkjon** [comment]
This does appear to have fixed the redemption issue as well.

**Pull request merged**